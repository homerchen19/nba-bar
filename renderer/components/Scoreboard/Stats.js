import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';
import { List } from 'react-virtualized';

import { colors } from '@styles/theme';

const Wrapper = styled(Flex)`
  width: 100%;
  padding: 5px 0;
  background: #fff;
`;

const StyledCell = styled(Flex.Item)`
  padding: 8px 26px;
  border-bottom: 1px solid ${colors.white};
  text-align: center;
`;

const teamStatsKeys = [
  { abbr: 'FG%', key: 'field_goals_percentage' },
  { abbr: '3P%', key: 'three_pointers_percentage' },
  { abbr: 'FT%', key: 'free_throws_percentage' },
  { abbr: 'AST', key: 'assists' },
  { abbr: 'REB', key: '' },
  { abbr: 'STL', key: 'steals' },
  { abbr: 'BLK', key: 'blocks' },
  { abbr: 'TOV', key: 'turnovers' },
  { abbr: 'FOUL', key: 'fouls' },
];

const Stats = ({ home, visitor }) => (
  <Wrapper justify="center">
    <List
      width={300}
      height={340}
      rowCount={teamStatsKeys.length}
      rowHeight={340 / teamStatsKeys.length}
      rowRenderer={({ key, index, style }) => {
        const { abbr, key: statsKey } = teamStatsKeys[index];

        return (
          <Flex key={key} style={style} justify="center">
            <StyledCell key="home">
              {abbr === 'REB'
                ? +home.stats.rebounds_offensive +
                  +home.stats.rebounds_defensive
                : home.stats[statsKey]}
            </StyledCell>
            <StyledCell key="item">
              <b>{abbr}</b>
            </StyledCell>
            <StyledCell key="visitor">
              {abbr === 'REB'
                ? +visitor.stats.rebounds_offensive +
                  +visitor.stats.rebounds_defensive
                : visitor.stats[statsKey]}
            </StyledCell>
          </Flex>
        );
      }}
    />
  </Wrapper>
);

Stats.propTypes = {
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
};

export default Stats;
