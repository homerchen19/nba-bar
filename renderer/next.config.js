const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
