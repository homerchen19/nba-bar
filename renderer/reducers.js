import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
