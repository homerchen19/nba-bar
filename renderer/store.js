import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
