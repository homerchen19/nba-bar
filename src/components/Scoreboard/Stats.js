import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';
import { StickyTable, Row, Cell } from 'react-sticky-table';

import { colors } from '../../styles/theme';
import nba from '../../utils/nba';

const Wrapper = styled(Flex)`
  width: 100%;
  padding: 5px 0;
  background: #fff;
`;

const StyledCell = styled(Cell)`
  padding: 8px 26px;
  border-bottom: 1px solid ${colors.white};
  text-align: center;
`;

const teamStatsKeys = {
  'FG%': 'field_goals_percentage',
  '3P%': 'three_pointers_percentage',
  'FT%': 'free_throws_percentage',
  AST: 'assists',
  REB: '',
  STL: 'steals',
  BLK: 'blocks',
  TOV: 'turnovers',
  FOUL: 'fouls',
};

const reanderStatsRow = (homeStats, visitorStats) =>
  Object.keys(teamStatsKeys).map(key => (
    <Row key={key}>
      <StyledCell key="home">
        {key === 'REB'
          ? +homeStats.rebounds_offensive + +homeStats.rebounds_defensive
          : homeStats[teamStatsKeys[key]]}
      </StyledCell>
      <StyledCell key="item">
        <b>{key}</b>
      </StyledCell>
      <StyledCell key="visitor">
        {key === 'REB'
          ? +visitorStats.rebounds_offensive + +visitorStats.rebounds_defensive
          : visitorStats[teamStatsKeys[key]]}
      </StyledCell>
    </Row>
  ));

const Stats = ({ home, visitor }) => (
  <Wrapper justify="center">
    <StickyTable stickyColumnCount={0}>
      <Row key="header">
        <StyledCell
          key="home"
          style={{
            color: nba.getTeamBackgroundColor(home.name),
            fontWeight: '600',
          }}
        >
          {home.name}
        </StyledCell>
        <StyledCell />
        <StyledCell
          key="visitor"
          style={{
            color: nba.getTeamBackgroundColor(visitor.name),
            fontWeight: '600',
          }}
        >
          {visitor.name}
        </StyledCell>
      </Row>
      {reanderStatsRow(home.stats, visitor.stats)}
    </StickyTable>
  </Wrapper>
);

Stats.propTypes = {
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
};

export default Stats;
