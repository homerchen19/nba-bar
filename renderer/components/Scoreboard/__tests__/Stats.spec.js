import React from 'react';
import renderer from 'react-test-renderer';

import Stats from '../Stats';

const realMath = Object.create(global.Math);

beforeAll(() => {
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
});

afterAll(() => {
  global.Math = realMath;
});

it('be defined', () => {
  expect(Stats).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    home: {
      name: 'IND',
      stats: {
        assists: '32',
        blocks: '3',
        field_goals_attempted: '93',
        field_goals_made: '50',
        field_goals_percentage: '53.8',
        fouls: '24',
        free_throws_attempted: '15',
        free_throws_made: '11',
        free_throws_percentage: '73.3',
        full_timeout_remaining: '2',
        points: '126',
        rebounds_defensive: '34',
        rebounds_offensive: '13',
        short_timeout_remaining: '0',
        steals: '11',
        team_fouls: '21',
        team_rebounds: '6',
        team_turnovers: '1',
        technical_fouls: '1',
        three_pointers_attempted: '29',
        three_pointers_made: '15',
        three_pointers_percentage: '51.7',
        turnovers: '12',
      },
    },
    visitor: {
      name: 'GSW',
      stats: {
        assists: '29',
        blocks: '5',
        field_goals_attempted: '83',
        field_goals_made: '37',
        field_goals_percentage: '44.6',
        fouls: '14',
        free_throws_attempted: '24',
        free_throws_made: '23',
        free_throws_percentage: '95.8',
        full_timeout_remaining: '2',
        points: '106',
        rebounds_defensive: '26',
        rebounds_offensive: '13',
        short_timeout_remaining: '0',
        steals: '6',
        team_fouls: '12',
        team_rebounds: '2',
        team_turnovers: '0',
        technical_fouls: '0',
        three_pointers_attempted: '29',
        three_pointers_made: '9',
        three_pointers_percentage: '31.0',
        turnovers: '16',
      },
    },
  };
  const tree = renderer.create(<Stats {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
