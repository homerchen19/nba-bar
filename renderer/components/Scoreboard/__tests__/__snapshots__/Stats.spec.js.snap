// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="am-flexbox Stats__Wrapper-prv853-0 lmyAZB am-flexbox-justify-center am-flexbox-align-center"
>
  <div
    aria-label="grid"
    aria-readonly={true}
    className="ReactVirtualized__Grid ReactVirtualized__List"
    onScroll={[Function]}
    role="grid"
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "boxSizing": "border-box",
        "direction": "ltr",
        "height": 340,
        "overflowX": "hidden",
        "overflowY": "hidden",
        "position": "relative",
        "width": 300,
        "willChange": "transform",
      }
    }
    tabIndex={0}
  >
    <div
      className="ReactVirtualized__Grid__innerScrollContainer"
      role="rowgroup"
      style={
        Object {
          "height": 340,
          "maxHeight": 340,
          "maxWidth": 300,
          "overflow": "hidden",
          "pointerEvents": "",
          "position": "relative",
          "width": "auto",
        }
      }
    >
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          53.8
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            FG%
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          44.6
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 37.77777777777778,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          51.7
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            3P%
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          31.0
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 75.55555555555556,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          73.3
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            FT%
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          95.8
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 113.33333333333334,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          32
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            AST
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          29
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 151.11111111111111,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          47
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            REB
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          39
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 188.88888888888889,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          11
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            STL
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          6
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 226.66666666666666,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          3
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            BLK
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          5
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 264.44444444444446,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          12
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            TOV
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          16
        </div>
      </div>
      <div
        className="am-flexbox am-flexbox-justify-center am-flexbox-align-center"
        style={
          Object {
            "height": 37.77777777777778,
            "left": 0,
            "position": "absolute",
            "top": 302.22222222222223,
            "width": "100%",
          }
        }
      >
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          24
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          <b>
            FOUL
          </b>
        </div>
        <div
          className="am-flexbox-item Stats__StyledCell-prv853-1 kHsgoc"
        >
          14
        </div>
      </div>
    </div>
  </div>
</div>
`;
